File: parking-lot/go-without-multithreading/strategy/fee/vehicle_based_fee_strategy.go
package fee

import (
	"app/parking-lot/go-without-multithreading/entities"
	"app/parking-lot/go-without-multithreading/enums"
)

type VehicleBasedFeeStrategy struct{}

func (s *VehicleBasedFeeStrategy) CalculateFee(v entities.Vehicle) float64 {
	switch v.GetType() {
	case enums.Motorcycle: return 5.0
	case enums.Car: return 10.0
	case enums.Truck: return 20.0
	default: return 10.0
	}
}

File: parking-lot/java-without-multithreading/strategy/fee/VehicleBasedFeeStrategy.java
package strategy.fee;
import entities.Vehicle;

public class VehicleBasedFeeStrategy implements FeeStrategy {
    @Override
    public double calculateFee(Vehicle vehicle) {
        switch(vehicle.getType()) {
            case MOTORCYCLE: return 5.0;
            case CAR: return 10.0;
            case TRUCK: return 20.0;
            default: return 10.0;
        }
    }
}
